File: SPUIKitCategory/UIFont/UIFont+FastCreate.m
//
//  UIFont+FastCreate.m
//  SmartCity
//
//  Created by LSP on 2017/5/19.
//  Copyright © 2017年 LSP. All rights reserved.
//

#import "UIFont+FastCreate.h"

@implementation UIFont (CircleFastCreate)

+ (instancetype)regularPingFangOfSize:(CGFloat)size {
    UIFont *font = [self fontWithName:@"PingFangSC-Regular" size:size];
    return font ? font : [UIFont systemFontOfSize:size];
}
+ (instancetype)mediumPingFangOfSize:(CGFloat)size {
    UIFont *font = [self fontWithName:@"PingFangSC-Medium" size:size];
    return font ? font : [UIFont systemFontOfSize:size];
}
+ (instancetype)boldPingFangOfSize:(CGFloat)size {
    UIFont *font = [self fontWithName:@"PingFangSC-Semibold" size:size];
    return font ? font : [UIFont boldSystemFontOfSize:size];
}

@end

File: Sources/UIFont/UIFont+FastCreate.swift
//
//  UIFont+FastCreate.swift
//  SPUIKitCategory module
//
//  Created by LSP on 2020/12/22.
//  Copyright © 2020 LSP. All rights reserved.
//

#if !os(macOS)
import UIKit

extension UIFont {
    
    public
    class func regularPingFang(ofSize size: CGFloat) -> UIFont {
        if let f = UIFont(name: "PingFangSC-Regular", size: size) {
            return f
        }
        return UIFont.systemFont(ofSize: size)
    }
    
    public
    class func mediumPingFang(ofSize size: CGFloat) -> UIFont {
        if let f = UIFont(name: "PingFangSC-Medium", size: size) {
            return f
        }
        return UIFont.systemFont(ofSize: size)
    }
    
    public
    class func boldPingFang(ofSize size: CGFloat) -> UIFont {
        if let f = UIFont(name: "PingFangSC-Semibold", size: size) {
            return f
        }
        return UIFont.boldSystemFont(ofSize: size)
    }
}
#endif
